core/templates/dev/head/services/contextual/UrlServiceSpec.js

```diff
@@ -108,6 +108,17 @@ describe('Url Service', function() {
     }).toThrow();
   });
 
+  it('should correctly retrieve topic name from url', function() {
+    mockLocation.pathname = '/topic/abcdefgijklm';
+    expect(
+      UrlService.getTopicNameFromLearnerUrl()
+    ).toBe('abcdefgijklm');
+    mockLocation.pathname = '/topc/abcdefgijklm';
+    expect(function(){
+      UrlService.getTopicNameFromLearnerUrl();
+    }).toThrow();
```

Comment:
toThrowError, and check the error message.

Reply:
Done.